SO-6369: Fix handling of enums in revision conflict processor (9.x)

AAAlinaaa opened the pull request.

AAAlinaaa committed:
SO-6369: Ensure enums aren't handled as objects

AAAlinaaa committed:
SO-6369: Add test for enum property type conflict handling

apeteri commented:
Ports changes in https://github.com/b2ihealthcare/snow-owl/pull/1358 to 9.x.

apeteri reviewed (approved):
👍

nagyo reviewed (approved):
LGTM with the same note from the sibling PR

apeteri committed:
fix(index.tests): Verify that a consistent enum change on both...

sides can be rebased successfully and the value remains the same
following the rebase.

nagyo reviewed (approved):
LGTM!

cmark commented:
Late LGTM!
